server.py

```diff
@@ -214,13 +217,107 @@ def _parse_languages(languages: str) -> Dict[str, int]:
     return ret
 
 
+def _parse_experience(experience: str) -> List[int]:
+    """
+    Return a string containing integers separated by commas as a list of ints.
+    """
+
+    ret = list()
+    if not experience:
+        return ret
+    for experience_level in experience.split(","):
+        ret.append(int(experience_level))
+    return ret
+
+
+def _get_coach_json(coach_args: JSON) -> CoachJSON:
+    """
+    Return a CoachJSON object corresponding the args from the get coaches
+    request.
+    """
+
+    # Turn all interger arguments into ints.
+    for int_arg in ["birth_year"]:
+        if int_arg in coach_args:
+            coach_args[int_arg] = int(coach_args[int_arg])
+
+    # Turn all list of interger arguments into lists of ints
+    for experience_args in ["need", "rights", "housing"]:
+        if experience_args in coach_args:
+            coach_args[experience_args] = _parse_experience(
+                coach_args[experience_args]
+            )
+
+    # Special case languages as it's a dict.
+    if "languages" in coach_args:
+        coach_args["languages"] = _parse_languages(coach_args["languages"])
+    
+    return coach_args
+
+
+def _include_coach(coach: CoachJSON, coach_filter: CoachJSON) -> bool:
+    """
+    Check whether to include a coach from the db in a repsoonse by filtering on
+    if the coach matches the fields in coach_filter.
+    """
+
+    if not coach_filter:
+        return True
+
+    # Keys where we need coach[key] == coach_filter[key]
+    direct_comparison_keys = [
+        "available",
+        "bio",
+        "birth_year",
+        "gender",
+        "id",
+        "name",
+    ]
+    for key in direct_comparison_keys:
+        if key in coach_filter:
+            if not coach.get(key) == coach_filter[key]:
```

Comment:
These two `if`s can be collapsed into `if coach.get(key) != coach_filter.get(key, object())` (where `object()` is used as a default value that will not match anything except that exact object instance).